server/src/main/java/org/opensearch/cluster/routing/remote/InternalRemoteRoutingTableService.java

```diff
@@ -362,6 +203,16 @@ protected void doStart() {
         final Repository repository = repositoriesService.get().repository(remoteStoreRepo);
         assert repository instanceof BlobStoreRepository : "Repository should be instance of BlobStoreRepository";
         blobStoreRepository = (BlobStoreRepository) repository;
+        compressor = blobStoreRepository.getCompressor();
+
+        this.remoteIndexRoutingTableStore = new RemoteRoutingTableBlobStore<>(
+            new BlobStoreTransferService(blobStoreRepository.blobStore(), threadPool),
+            blobStoreRepository,
+            clusterName,
+            threadPool,
+            ThreadPool.Names.REMOTE_STATE_READ,
+            clusterSettings
```

Comment:
Should we have a remote routing thread pool, in case we want to enable them separately?

Reply:
We haven't yet added the hybrid support, we either enable both or neither of them. Should we add separate pools when we add hybrid mode support?

Reply:
I don' think it is needed at the moment since the remote routing and remote state are clubbed together under Publication. Moreover we can take this up if we see any issue when we do load testing of the feature.